src/main/java/net/minecraftforge/common/capabilities/CapabilityManager.java

```diff
@@ -59,10 +59,26 @@ <T> Capability<T> get(String realName, boolean registering)
     }
 
     // INTERNAL
+    private static final Type AUTO_REGISTER = Type.getType(AutoRegisterCapability.class);
     private final IdentityHashMap<String, Capability<?>> providers = new IdentityHashMap<>();
     public void injectCapabilities(List<ModFileScanData> data)
     {
         var event = new RegisterCapabilitiesEvent();
+
+        var autos = data.stream()
+            .flatMap(e -> e.getAnnotations().stream())
+            .filter(a -> AUTO_REGISTER.equals(a.annotationType()))
+            .map(a -> a.clazz())
+            .distinct()
+            .sorted((a,b) -> a.toString().compareTo(b.toString()))
```

Comment:
Some lambda to method reference cleanup.
```suggestion
            .map(ModFileScanData.AnnotationData::clazz)
            .distinct()
            .sorted(Comparator.comparing(Type::toString))
```

Reply:
I'll take the Comparator one, but I like the clazz one as it's shorter. And IMO that's 1/2 the power of lambdas.